consensus/src/rand/rand_gen/rand_store.rs

```diff
@@ -42,18 +42,26 @@ impl<S: Share> ShareAggregator<S> {
     }
 
     pub fn try_aggregate(
-        &self,
+        self,
         rand_config: &RandConfig,
         rand_metadata: RandMetadata,
-    ) -> Option<Randomness> {
+        decision_tx: Sender<Randomness>,
+    ) -> Either<Self, RandShare<S>> {
         if self.total_weight < rand_config.threshold_weight() {
-            return None;
+            return Either::Left(self);
         }
-        Some(S::aggregate(
-            self.shares.values(),
-            rand_config,
-            rand_metadata,
-        ))
+        let rand_config = rand_config.clone();
+        let self_share = self
+            .get_self_share()
+            .expect("Aggregated item should have self share");
+        tokio::task::spawn_blocking(move || {
```

Comment:
[here](https://dtantsur.github.io/rust-openstack/tokio/sync/mpsc/struct.UnboundedSender.html#method.send) it says it never blocks... so no need of `spawn_blocking`?

Reply:
oh, i mean `decision_tx.send`

Reply:
nvm, it's for the aggregation